### PR TITLE
Sketchbook prototype

### PR DESCRIPTION
**Pull request by ptomato**
Here's what I had for the Sketchbook app prototype. Not sure if it even makes sense to merge it, but I thought I should at least put it in a branch.

### CODE REVIEW

File: src/codeview.js [hunk 1]
```diff
@@ -232,6 +239,7 @@ var Codeview = GObject.registerClass({
             });
             this._cached_ast = null;
             this.compile(false);
+            this.notify('text');
```
Comment (manuq) [inline src/codeview.js:242]: The notify is automatic at least in python, unless you add the`GObject.ParamFlags.EXPLICIT_NOTIFY` flag to the property. Isn't the same in GJS?

File: src/codeview.js [hunk 2]
```diff
@@ -223,6 +227,9 @@ var Codeview = GObject.registerClass({
     }
 
     set text(value) {
+        if (this._buffer && this._buffer.text === value)
+            return;
```
Comment (manuq) [inline src/codeview.js:231]: Maybe you need the `GObject.ParamFlags.EXPLICIT_NOTIFY` flag for this to return without notifying?

File: src/app.js [hunk 3]
```diff
@@ -30,6 +30,8 @@ function _toolboxClassForAppId(targetAppId) {
         return imports.OperatingSystemApp.toolbox.OSToolbox;
     case 'com.endlessm.Sidetrack':
         return imports.RileyMaze.toolbox.RMZToolbox;
+    case 'name.ptomato.AbsentRhythm':
```
Comment (manuq) [inline src/app.js]: Now this could be the actual name, 'com.endlessm.Sketchbook'.

File: src/sketchbook/model.js [hunk 4]
```diff
@@ -1,29 +1,142 @@
 /* exported SketchModel */
 
-const {GObject} = imports.gi;
+const {Gio, GObject} = imports.gi;
+
+const INTERFACE = `
+<node>
+  <interface name="name.ptomato.AbsentRhythm.Variables">
```
Comment (manuq) [inline src/sketchbook/model.js]: This need to be 'com.endlessm.Sketchbook' now.

File: src/sketchbook/model.js [hunk 5]
```diff
@@ -1,29 +1,142 @@
...
-            ''),
+            GObject.ParamFlags.READWRITE, ''),
     },
 }, class SketchModel extends GObject.Object {
+    // This ought to be implemented via the Gio.AsyncInitable interface, but
+    // that's not currently possible in GJS
+    async initAsync(cancellable = null) {
+        const sessionBus = await Gio.DBus.get(Gio.BusType.SESSION, cancellable);
+        this._proxy = await new Promise((resolve, reject) => {
+            void new VariablesProxy(sessionBus, 'name.ptomato.AbsentRhythm',
```
Comment (manuq) [inline src/sketchbook/model.js]: Ditto, rename needed.

**Review by manuq — CHANGES_REQUESTED** [review]
Looking good but I couldn't test it yet. Leaving a few comments to ;keep the rock rolling.

File: src/codeview.js (hunk 1 above)
Comment (ptomato) [reply src/codeview.js:242]: It's automatic if you use [`GObject.set_property()`](https://gjs-docs.gnome.org/gobject20~2.0_api/gobject.object#method-set_property), but manual if you use JS property setters.

Comment (manuq) [reply src/codeview.js:242]: oh. Ok.

**Review by manuq — APPROVED** [review]
Thanks for clarifying and the rename. I haven't tried it but I'll merge as-is since you are moving to the Platform team, and there are no plans to this prototype right now, but yes in the future.

**Pull request merged**